extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java

```diff
@@ -232,20 +233,12 @@ private void removeNoRepositoryBeanClasses(List<ClassInfo> interfacesExtendingRe
     }
 
     private Collection<DotName> getAllNoRepositoryBeanInterfaces(IndexView index) {
-        Set<DotName> result = new HashSet<>();
-        Collection<ClassInfo> knownClasses = index.getKnownClasses();
-        for (ClassInfo clazz : knownClasses) {
-            if (!Modifier.isInterface(clazz.flags())) {
-                continue;
-            }
-            boolean found = false;
-            for (ClassInfo classInfo : knownClasses) {
-                if (classInfo.classAnnotation(DotNames.SPRING_DATA_NO_REPOSITORY_BEAN) != null) {
-                    result.add(classInfo.name());
-                }
-            }
-        }
-        return result;
+        return index.getKnownClasses()
+                .stream()
+                .filter(classInfo -> Modifier.isInterface(classInfo.flags()))
+                .filter(classInfo -> classInfo.classAnnotation(DotNames.SPRING_DATA_NO_REPOSITORY_BEAN) != null)
```

Comment:
I might be missing something but why don't we search for the annotation instead? And then filter the target to be a class and then check the interfaces?

Reply:
@johnaohara I pushed another commit with an implementation using the index. Can you check it's good for you? I think it should be better as we don't end up scanning the whole index.

Reply:
@gsmet I was just testing a change similar to what you pushed. It looks good, but I'll pull down your changes and verify against the start up test

Reply:
It is good,  CPU time in this method has gone from 24.7% in 2.1.1.Final to 0.4% with this PR